### PR TITLE
feat: simplify the docker-compose setup so we do less version coordinations

### PR DESCRIPTION
**Pull request by fuziontech**
- feat: remove version from docker compose to support new spec
- feat: simplify the docker-compose setup so we do less version coordinations

depends on https://github.com/PostHog/posthog/pull/12997

This moves from `docker-compose` to `docker compose` which is how github differentiates between v1 and v2

## Problem

This makes it easier by having a base docker-compose template for development by using `extends`. Very exciting!
https://github.com/compose-spec/compose-spec/blob/master/spec.md#extends

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by fuziontech** [comment]
Ignore the hobby CI. It's broken and needs fixing (there is a prompt for input while installing dependencies now)

File: docker-compose.base.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#
+# docker-compose file used ONLY for local development.
```
Comment (guidoiaquinti) [inline docker-compose.base.yml:2]: Not sure if this comment is accurate

**Review by guidoiaquinti — APPROVED** [review]
Minor comment, overall LGTM

File: docker-compose.base.yml (hunk 1 above)
Comment (fuziontech) [reply docker-compose.base.yml:2]: ah, good catch!

**Pull request merged**